Screens/NewRace.js

```diff
@@ -0,0 +1,170 @@
+import React, { Component } from 'react'
+import {
+  Alert,
+  Button,
+  Picker,
+  StyleSheet,
+  Text,
+  TextInput,
+  View
+} from 'react-native'
+
+class NewRace extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      codex: '',
+      gender: 'Female',
+      temperature: '',
+      precipitation: 'None',
+      course: '',
+      racers: 6,
+      position: 'Start Gate',
+    }
+  }
+
+  static navigationOptions = {
+    title: 'New Race'
+  }
+
+  _handleBtnPress() {
+    const { navigate } = this.props.navigation
+    if (this.state.codex       === '' ||
+        this.state.temperature === '' ||
+        this.state.course      === ''
+       ) {
+      Alert.alert('Error: please enter a value for all text fields.')
+    }
```

Comment:
You can avoid the need for this check by setting the `disabled` prop on the button (i.e., you can only press it when the fields are non-empty).